### PR TITLE
Build refactor to webpack

### PR DESCRIPTION
**Pull request by larkin**
Refactor the build process to use webpack and support vue SFCs.  Also traded jshint + jscs for eslint, and tweaked linter rules based on discussion.

Gulp has been removed, use `npm run` to run various tasks.

This changeset will merge into topic-core-refactor to allow concurrent development efforts on our various refactoring tasks while protecting master from instability.

# Author Checklist

1. Changes address original issue? Y, as discussed
2. Unit tests included and/or updated with changes? Y, as possible
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

Remaining todos to track in topic refactor (not to be completed in this pull):

* [ ] Verify build constants are still working
* [ ] Ensure compatibility with libraries packaged with webpack

### CODE REVIEW

File: example/eventGenerator/src/EventTelemetry.js [hunk 1]
```diff
@@ -22,24 +22,24 @@
...
-    function (transcript) {
+    ['../data/transcript.json'],
+    function (messages) {
         "use strict";
 
-        var firstObservedTime = Date.now(),
-            messages = JSON.parse(transcript);
-        
+        var firstObservedTime = Date.now();
+            // messages = JSON.parse(transcript);
```
Comment (akhenry) [inline example/eventGenerator/src/EventTelemetry.js]: Let's remove this line now that webpack is magically providing parsed javascript objects.

File: example/styleguide/bundle.js [hunk 2]
```diff
@@ -1,10 +1,26 @@
 define([
     "./src/ExampleStyleGuideModelProvider",
     "./src/MCTExample",
+    "text!./res/templates/intro.html",
```
Comment (akhenry) [inline example/styleguide/bundle.js]: `text!` still necessary here?

File: karma.conf.js [hunk 3]
```diff
@@ -104,8 +74,19 @@ module.exports = function(config) {
             foldAll: false
         },
 
-        // Continuous Integration mode.
-        // If true, Karma captures browsers, runs the tests and exits.
+        preprocessors: {
+            // add webpack as preprocessor
+            'platform/**/*Spec.js': [ 'webpack' ],
```
Comment (akhenry) [inline karma.conf.js]: Coverage?

File: karma.conf.js [hunk 4]
```diff
@@ -104,8 +74,19 @@ module.exports = function(config) {
             foldAll: false
         },
 
-        // Continuous Integration mode.
-        // If true, Karma captures browsers, runs the tests and exits.
+        preprocessors: {
+            // add webpack as preprocessor
```
Comment (akhenry) [inline karma.conf.js]: no explicit preprocessor needed for coverage now?

File: platform/commonUI/mobile/test/DeviceClassifierSpec.js [hunk 5]
```diff
@@ -70,11 +70,11 @@ define(
...
             TEST_PERMUTATIONS.forEach(function (trueMethods) {
                 var summary = trueMethods.length === 0 ?
-                        "device has no detected characteristics" :
-                        "device " + (trueMethods.join(", "));
+                    "device has no detected characteristics" :
+                    "device " + (trueMethods.join(", "));
 
                 describe("when " + summary, function () {
-                    var classifier;
+                    var classifier; // eslint-disable-line
```
Comment (akhenry) [inline platform/commonUI/mobile/test/DeviceClassifierSpec.js]: Seen this in a few places. Why does the linter not like this? Uninitialized variable?

File: src/plugins/summaryWidget/src/eventHelpers.js [hunk 6]
```diff
@@ -12,7 +11,7 @@ define([], function () {
                 event: event,
                 callback: callback,
                 context: context,
-                _cb: !!context ? callback.bind(context) : callback
```
Comment (akhenry) [inline src/plugins/summaryWidget/src/eventHelpers.js]: Does the linter disallow `!!`? If so, good.

File: example/styleguide/bundle.js (hunk 2 above)
Comment (larkin) [reply example/styleguide/bundle.js]: Yes but we can do even better with html-loader.  Updated.

**Comment by larkin** [comment]
@akhenry responded to your comments and fixed some of the things you noted.  Back to you!

**Comment by akhenry** [comment]
## Reviewer Checklist

* Changes appear to address issue? Y
* Appropriate unit tests included? N/A Build configuration. Test configuration has been updated for compatibility though.
* Code style and in-line documentation are appropriate? Y
* Commit messages meet standards? Y

**Pull request merged**